UNRELEASED.md

```diff
@@ -29,3 +29,12 @@ and ensure that the release notes and documentation has been included as part of
 ## Bugfixes
 
 ## What’s New
+
+* The type class `HasField` (methods `getField` and `setField`) has been split
+  in two: `GetField` and `SetField`, each one with the correspondingly named method.
+
+* User-defined instances of the class `GetField` enable the use of dot-syntax
+  for field access, i.e. `rec.field`
+
+* User-defined instances of the class `SetField` enable the use of with-syntax
```

Comment:
Could you add a small example on how to migrate an existing user defined method that uses HasField to GetFIeld/SetFIeld? Feel free to do this in a separate PR.

Reply:
Added in a new commit